hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/auth/ITestAssumeRole.java

```diff
@@ -448,7 +448,7 @@ public void testReadOnlyOperations() throws Throwable {
         policy(
             statement(false, S3_ALL_BUCKETS, S3_PATH_WRITE_OPERATIONS),
             STATEMENT_ALL_S3,
-            STATEMENT_ALLOW_SSE_KMS_READ));
+            STATEMENT_ALLOW_SSE_KMS_RW));
```

Comment:
yes, think it may have always been broken when CSE is enabled. Without RW access, encryption client is not able to generate the key required to encrypt for a PUT. So the test fails with a "unable to generate" key exception.

Reply:
Oh yes... We also saw this error and this is the fix. 
CC @mehakmeet